File: VNASchedule/VNASchedule/VNASchedule/Bean/DeviceInfo.cs
using System;
namespace VNASchedule.Bean
{
    public class DeviceInfo
    {
        public string DeviceId { get; set; }
        public string DevicePushToken { get; set; }
        public short DeviceOS { get; set; } // 1: Android   2: IOS  4: WindowPhone
        public string AppVersion { get; set; }
        public string DeviceOSVersion { get; set; }
        public string DeviceModel { get; set; }
    }
}

File: lib/base/model/device_info.dart
import 'package:json_annotation/json_annotation.dart';

part 'device_info.g.dart';

@JsonSerializable()
class DeviceInfo {
  late String DeviceId;
  late String? DevicePushToken;
  late int DeviceOS;
  late String AppVersion;
  late String DeviceOSVersion;
  late String DeviceModel;

  DeviceInfo(this.DeviceId, this.DevicePushToken, this.DeviceOS,
      this.AppVersion, this.DeviceOSVersion, this.DeviceModel);

  DeviceInfo.required(
      {required this.DeviceId,
      required this.DevicePushToken,
      required this.DeviceOS,
      required this.AppVersion,
      required this.DeviceOSVersion,
      required this.DeviceModel});

  factory DeviceInfo.fromJson(Map<String, dynamic> json) =>
      _$DeviceInfoFromJson(json);

  Map<String, dynamic> toJson() => _$DeviceInfoToJson(this);
}
